onnxruntime/core/framework/bfc_arena.h

```diff
@@ -444,16 +475,21 @@ class BFCArena : public IArenaAllocator {
   // newly-created chunk.
   int64_t next_allocation_id_;
 
+  // Counter tracking total number of allocation regions
+  int64_t allocation_region_counter_;
```

Comment:
Any reason to track this separately? Can RegionManager not give you this info (RegionManager::regions_.size())?

Reply:
The idea was to be able to detect the "initial" allocation and not de-allocate it even if it is totally unused so as to avoid re-allocation on every Run. The only allocation regions that are considered for shrinkages are the second allocation onwards and I stamp every allocation region with this metadata so as to be quickly recognize the first allocation and not consider it for de-allocation when Shrink is called. I suppose I can achieve the same thing by ignoring the first allocation stored in RegionManager::regions_. I ll have to think if there are circumstances where-in the first element in RegionManager::regions_ is not the initial allocation. Either way, I felt this approach was cleaner. Let me know if you feel differently and I will think of the alternate approach.